### PR TITLE
Fix getting pyc mtime on Python 3.7

### PR DESCRIPTION
**Pull request by hroncok**
Fixes https://github.com/rpm-software-management/rpmlint/issues/128

### CODE REVIEW

File: FilesCheck.py [hunk 1]
```diff
@@ -379,6 +379,25 @@ def py_demarshal_long(b):
...
+
+
+def pyc_mtime_from_chunk(chunk):
+    """From given chunk (beginning of the file), return mtime or None
+
+    From Python 3.7, mtime is not always present.
+    """
+    magic = pyc_magic_from_chunk(chunk)
+    second = py_demarshal_long(chunk[4:8])
+    if magic >= 3390:  # Python 3.7
```
Comment (encukou) [inline FilesCheck.py]: Maybe you could add a link to the specification, i.e. https://www.python.org/dev/peps/pep-0552/#specification

File: FilesCheck.py [hunk 2]
```diff
@@ -379,6 +379,27 @@ def py_demarshal_long(b):
...
+    """From given chunk (beginning of the file), return Python magic number"""
+    return py_demarshal_long(chunk[:4]) & 0xffff
+
+
+def pyc_mtime_from_chunk(chunk):
+    """From given chunk (beginning of the file), return mtime or None
+
+    From Python 3.7, mtime is not always present.
+
+    See https://www.python.org/dev/peps/pep-0552/#specification
```
Comment (hroncok) [inline FilesCheck.py]: @encukou added here

File: FilesCheck.py [hunk 3]
```diff
@@ -379,6 +379,27 @@ def py_demarshal_long(b):
...
+    """
+    magic = pyc_magic_from_chunk(chunk)
+    second = py_demarshal_long(chunk[4:8])
+    if magic >= _python_magic_values['3.7'][0]:
```
Comment (hroncok) [inline FilesCheck.py]: changed to not hardcode the magic number here again.

File: test/pyc/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+!*.pyc
```
Comment (hroncok) [inline test/pyc/.gitignore]: Forgot to add the pyc files, as they were gitignored (obviously), so added them now together with this.

File: FilesCheck.py (hunk 1 above)
Comment (hroncok) [reply FilesCheck.py]: added

**Comment by Conan-Kudo** [comment]
This looks pretty good! I'll try to do some testing on something with Python 3.7 to be sure it doesn't break anything and then I'll merge it, if there are no issues.

**Comment by hroncok** [comment]
bump?

**Comment by Conan-Kudo** [comment]
Will take a look later today. I didn't have anything with Python 3.7 to use it with until recently.

**Comment by Conan-Kudo** [comment]
Looks good to me, merging!

**Comment by Conan-Kudo** [comment]
Thanks!

**Pull request merged**